libsolidity/analysis/ConstantEvaluator.cpp

```diff
@@ -46,9 +48,24 @@ void ConstantEvaluator::endVisit(BinaryOperation const& _operation)
 	if (!dynamic_cast<RationalNumberType const*>(rightType.get()))
 		m_errorReporter.fatalTypeError(_operation.rightExpression().location(), "Invalid constant expression.");
 	TypePointer commonType = leftType->binaryOperatorResult(_operation.getOperator(), rightType);
-	if (Token::isCompareOp(_operation.getOperator()))
```

Comment:
This is a slight change in semantics. Did you test that against e.g. `bool constant x = 2**700 > 2**600;`?

Reply:
Fails, but in the non-constant case too:

```
3107.sol:2:21: Error: Operator > not compatible with types int_const 5260135901548373507240989882880128665550339802823173859498280903068732154297080822113666536277588451226982968856178217713019432250183803863127814770651880849955223671128444598191663757884322717271293251735781376 and int_const 4149515568880992958512407863691161151012446232242436899995657329690652811412908146399707048947103794288197886611300789182395151075411775307886874834113963687061181803401509523685376
  bool constant x = 2**700 > 2**600;
                    ^-------------^
```

Before this change it works for constants but not anything else. I think that should be consistent (ie. working in both cases).

Reply:
ok!

Reply:
Wait a second, this code is not even used for such constants. It is only used to determine the length of an array.

Reply:
yes